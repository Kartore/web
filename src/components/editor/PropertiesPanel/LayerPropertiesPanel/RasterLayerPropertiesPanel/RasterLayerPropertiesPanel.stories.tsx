import type { Meta, StoryObj } from '@storybook/react';

import { RasterLayerPropertiesPanel } from './index.ts';

const meta = {
	component: RasterLayerPropertiesPanel,
} satisfies Meta<typeof RasterLayerPropertiesPanel>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Default: Story = {
	args: {
		layer: {
			id: 'raster-layer-test',
			type: 'raster',
			source: 'openmaptiles',
			'source-layer': 'source-layer-test',
			minzoom: 0,
			maxzoom: 24,
			paint: {
				'raster-opacity': 0.5,
			},
		},
		sources: {
			openmaptiles: {
				type: 'vector',
				url: 'https://tileserver.kartore.io/tiles/openstreetmap/tile.json',
			},
			natural_earth_shaded_relief: {
				maxzoom: 6,
				tileSize: 256,
				tiles: [
					'https://kartore.github.io/naturalearthtiles/tiles/natural_earth_2_shaded_relief.raster/{z}/{x}/{y}.png',
				],
				type: 'raster',
			},
		},
	},
};
