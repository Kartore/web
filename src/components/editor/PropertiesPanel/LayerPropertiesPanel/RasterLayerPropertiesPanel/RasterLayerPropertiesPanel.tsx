import type { ComponentProps, FC } from 'react';

import type {
  RasterLayerSpecification,
  SourceSpecification,
} from '@maplibre/maplibre-gl-style-spec';

import { GeneralProperties } from '~/components/editor/PropertiesPanel/LayerPropertiesPanel/common/GeneralProperties';
import { RawDataProperties } from '~/components/editor/PropertiesPanel/LayerPropertiesPanel/common/RawDataProperties';
import type { onChangeType } from '~/components/editor/PropertiesPanel/LayerPropertiesPanel/utils/LayerUtil/LayerUtil.ts';
import { cn } from '~/utils/tailwindUtil.ts';

export type RasterLayerPropertiesPanelProps = Omit<ComponentProps<'div'>, 'onChange'> & {
  layer: RasterLayerSpecification;
  sources: { [key: string]: SourceSpecification };
  onChange?: onChangeType;
};

export const RasterLayerPropertiesPanel: FC<RasterLayerPropertiesPanelProps> = ({
  children,
  layer,
  sources,
  onChange,
  className,
  ...props
}) => {
  return (
    <div {...props} className={cn('flex flex-col gap-2', className)}>
      <GeneralProperties layer={layer} sources={sources} onChange={onChange} />
      <RawDataProperties layer={layer} />
      {children}
    </div>
  );
};

RasterLayerPropertiesPanel.displayName = 'RasterLayerPropertiesPanel';
