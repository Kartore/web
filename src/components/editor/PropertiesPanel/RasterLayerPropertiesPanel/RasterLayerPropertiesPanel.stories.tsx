import { RasterLayerPropertiesPanel } from '.';
import { Meta, StoryObj } from '@storybook/react';

const meta = {
  component: RasterLayerPropertiesPanel,
} satisfies Meta<typeof RasterLayerPropertiesPanel>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {},
};
