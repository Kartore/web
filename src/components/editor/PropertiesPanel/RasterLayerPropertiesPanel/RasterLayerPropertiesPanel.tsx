import { Box, BoxProps, forwardRef } from '@chakra-ui/react';
import { LayerSpecification } from '@maplibre/maplibre-gl-style-spec';
import { onChangeType } from '~/components/editor/PropertiesPanel/utils/LayerUtil/LayerUtil.ts';

export type RasterLayerPropertiesPanelProps = Omit<BoxProps, 'onChange' | 'children'> & {
  layer: LayerSpecification;
  onChange?: onChangeType;
};

export const RasterLayerPropertiesPanel = forwardRef<RasterLayerPropertiesPanelProps, 'div'>(
  ({ ...props }, ref) => {
    return <Box ref={ref} {...props}></Box>;
  }
);
