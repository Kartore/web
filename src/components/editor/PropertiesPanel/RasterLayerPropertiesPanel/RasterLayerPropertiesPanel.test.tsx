import { describe } from 'vitest';
import { render } from '@testing-library/react';
import { composeStories } from '@storybook/react';

import * as Stories from './RasterLayerPropertiesPanel.stories';

const { Default } = composeStories(Stories);

describe('Component: RasterLayerPropertiesPanel', () => {
  describe('Snapshot', () => {
    it('Default', () => {
      const { asFragment } = render(<Default />);
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
