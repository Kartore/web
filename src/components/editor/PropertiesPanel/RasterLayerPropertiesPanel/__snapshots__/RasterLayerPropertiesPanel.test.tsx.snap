// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Component: RasterLayerPropertiesPanel > Snapshot > Default 1`] = `
<DocumentFragment>
  <div />
</DocumentFragment>
`;
